File: Contract.Architecture.Backends/Contract.Architecture.Backend.Core/Contract/Logic/Modules/Kundenstamm/Kunden/DTOs/IKundeDetail.cs
using Contract.Architecture.Backend.Core.Contract.Logic.Modules.Bankwesen.Banken;
using System;

namespace Contract.Architecture.Backend.Core.Contract.Logic.Modules.Kundenstamm.Kunden
{
    public interface IKundeDetail
    {
        Guid Id { get; set; }

        string Name { get; set; }

        int Balance { get; set; }

        IBank Bank { get; set; }
    }
}

File: Contract.Architecture.Frontends/Contract.Architecture.Web.Core/src/app/model/kundenstamm/kunden/dtos/i-kunde-detail.ts
import { Bank, IBank } from 'src/app/model/bankwesen/banken/dtos/i-bank';
import { ApiKundeDetail } from './api/api-kunde-detail';

export interface IKundeDetail {
    id: string;
    name: string;
    balance: number;
    bank: IBank;
}

export abstract class KundeDetail {
    public static fromApiKundeDetail(apiKundeDetail: ApiKundeDetail): IKundeDetail {
        return {
            id: apiKundeDetail.id,
            name: apiKundeDetail.name,
            balance: apiKundeDetail.balance,
            bank: Bank.fromApiBank(apiKundeDetail.bank),
        };
    }
}
